python/dgl/model_zoo/chem/layers.py

```diff
@@ -0,0 +1,289 @@
+        """
+        Randomly init the element embeddings.
+        Args:
+            dim: the dim of embeddings
+            type_num: the largest atomic number of atoms in the dataset
+            pre_train: the pre_trained embeddings
+        """
+        super().__init__()
+        self._dim = dim
+        self._type_num = type_num
+        if pre_train is not None:
+            self.embedding = nn.Embedding.from_pretrained(pre_train,
+                                                          padding_idx=0)
+        else:
+            self.embedding = nn.Embedding(type_num, dim, padding_idx=0)
+
+    def forward(self, g, p_name="node"):
+        """Input type is dgl graph"""
+        atom_list = g.ndata["node_type"]
+        g.ndata[p_name] = self.embedding(atom_list)
+        return g.ndata[p_name]
+
+
+class EdgeEmbedding(nn.Module):
+    """
+    Convert the edge to embedding.
+    The edge links same pair of atoms share the same initial embedding.
+    """
+
+    def __init__(self, dim=128, edge_num=3000, pre_train=None):
+        """
+        Randomly init the edge embeddings.
+        Args:
+            dim: the dim of embeddings
+            edge_num: the maximum type of edges
+            pre_train: the pre_trained embeddings
+        """
+        super().__init__()
+        self._dim = dim
+        self._edge_num = edge_num
+        if pre_train is not None:
+            self.embedding = nn.Embedding.from_pretrained(pre_train,
+                                                          padding_idx=0)
+        else:
+            self.embedding = nn.Embedding(edge_num, dim, padding_idx=0)
+
+    def generate_edge_type(self, edges):
+        """
+        Generate the edge type based on the src&dst atom type of the edge.
+        Note that C-O and O-C are the same edge type.
+        To map a pair of nodes to one number, we use an unordered pairing function here
+        See more detail in this disscussion:
+        https://math.stackexchange.com/questions/23503/create-unique-number-from-2-numbers
+        Note that, the edge_num should larger than the square of maximum atomic number
+        in the dataset.
+        """
+        atom_type_x = edges.src["node_type"]
+        atom_type_y = edges.dst["node_type"]
+
+        return {
+            "type":
+            atom_type_x * atom_type_y +
+            (th.abs(atom_type_x - atom_type_y) - 1)**2 / 4
+        }
+
+    def forward(self, g, p_name="edge_f"):
+        g.apply_edges(self.generate_edge_type)
+        g.edata[p_name] = self.embedding(g.edata["type"])
+        return g.edata[p_name]
+
+
+class ShiftSoftplus(Softplus):
+    """
+    Shiftsoft plus activation function:
+        1/beta * (log(1 + exp**(beta * x)) - log(shift))
+    """
+
+    def __init__(self, beta=1, shift=2, threshold=20):
+        super().__init__(beta, threshold)
+        self.shift = shift
+        self.softplus = Softplus(beta, threshold)
+
+    def forward(self, input):
```

Comment:
`input` is a built-in function for python. Try to avoid using it for variable name.